src/access/DicomWebSeriesAccess.js

```diff
@@ -0,0 +1,140 @@
+import * as dicomweb from "../dicomweb.js";
+import { createDicomWebConfig, fixBulkDataURI } from "../utils/index.js";
+
+const nonFrameSOPs = {
+  "1.2.840.10008.5.1.4.1.1.88.11": "Basic Text SR",
+  "1.2.840.10008.5.1.4.1.1.88.22": "Enhanced SR",
+  "1.2.840.10008.5.1.4.1.1.88.33": "Comprehensive SR",
+  "1.2.840.10008.5.1.4.1.1.481.3": "RT Structure Set",
+  "1.2.840.10008.5.1.4.1.1.104.1": "Encapsulated PDF",
+};
+
+export class DicomWebSeriesAccess {
+  constructor(url, options, metadata) {
+    this.url = url;
+    this.options = options;
+    this.StudyInstanceUID = options.StudyInstanceUID;
+    this.metadata = metadata;
+
+    // Containers to store queried data
+    this.seriesInstanceUIDsMetadata = new Map();
```

Comment:
Don't assume that everything gets read up front, instead read things as they are requested, grouping them into maps of objects as required with an async function doing the await on each item as it arrives.